pysgrid/read_netcdf.py

```diff
@@ -150,11 +232,248 @@ def sgrid_compliant_file(self):
         
         """
         grid_vars = self.find_grid_topology_vars()
-        if len(grid_vars) > 0:
+        if grid_vars is not None:
             sgrid_compliant = True
         else:
             sgrid_compliant = False
         return sgrid_compliant
+    
+    
+class SGridND(object):
+    
+    topology_dim = None
+    
+    def __init__(self, sgrid, nc_dataset, topology_variable):
+        self._sgrid = sgrid
+        self.nc_dataset = nc_dataset
+        self.ncd = NetCDFDataset(self.nc_dataset)
+        self.topology_variable = topology_variable  # the netCDF variable with a cf_role of 'grid_topology'
+        self.topology_var = self.nc_dataset.variables[self.topology_variable]
+        self.pp = ParsePadding(self.topology_variable)
+        
+    @property
+    def sgrid(self):
+        return self._sgrid
+    
+    def set_dimensions(self):
+        ds_dims = self.nc_dataset.dimensions
+        grid_dims = [(ds_dim, len(ds_dims[ds_dim])) for ds_dim in ds_dims]
+        self._sgrid.dimensions = grid_dims
+        
+    def set_topology_dimension(self):
+        self._sgrid.topology_dimension = self.topology_dim
+    
+    def set_edge1_dimensions(self):
+        try:
+            edge1_dim = self.topology_var.edge1_dimensions
+            edge1_dim_padding = self.pp.parse_padding(edge1_dim)
+            self._sgrid.edge_1_dimension = edge1_dim
+            self._sgrid.edge_1_padding = edge1_dim_padding
+        except AttributeError:
+            pass
```

Comment:
There are a number of required and optional sgrid attributes that are defined per https://publicwiki.deltares.nl/display/NETCDF/Deltares+proposal+for+Staggered+Grid+data+model+(SGRID). For the optional attributes, I try to set an attribute for them. For instance, if a netCDF grid variable has a edge1_dimension, SGrid.edge_1_dimension is set, otherwise, SGrid.edge_1_dimension is None. In retrospect, SGrid.edge_1_dimension should be SGrid.edge1_dimension, but I can make that change relatively easily.